engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -353,7 +353,10 @@ Function resolveWarningFunction(
     try {
       selfWithoutWarnings = warnings.removeWarnings(self);
     } catch (UnsupportedMessageException e) {
-      throw CompilerDirectives.shouldNotReachHere(
+      CompilerDirectives.transferToInterpreter();
```

Comment:
No need for `transferToInterpreter`, as `raiseAssertionPanic` is behind Truffle boundary

Reply:
Well, `raiseAssertionPanic` is behind the boundary, but not the string concatenation. We have to avoid the string concatenation to be performed on fast path.